src/types/bchar.ml

```diff
@@ -0,0 +1,75 @@
+(* This file is free software, part of containers. See file "license" for more details. *)
+
+(** {1 Utils around char}
+
+    @since 0.14 *)
+
+include Char
+
+let equals (a:char) b = a=b
+let compare = Comparator.make Char.compare
+
+let lowercaseifAscii c =
+  if c >= 'A' && c <= 'Z'
+  then Char.unsafe_chr (Char. code c + 32)
+  else c
+
+let uppercaseifAscii c =
+  if c >= 'a' && c <= 'z'
+  then Char.unsafe_chr (Char.code c - 32)
+  else c
+
+let fromIntOrRaise = Char.chr
+let fromInt c = try Some (fromIntOrRaise c) with _ -> None
+let toInt = Char.code
+
+let isLowercase = function
+  | 'a' .. 'z' -> true
+  | _ -> false
+
+let isUppercase = function
+  | 'A' .. 'Z' -> true
+  | _ -> false
+
+let isPrint = function
+  | ' ' .. '~' -> true
+  | _ -> false
+
+let isWhitespace = function
+  | '\t'
+  | '\n'
+  | '\011' (* vertical tab *)
+  | '\012' (* form feed *)
+  | '\r'
+  | ' '
+    -> true
+  | _
+    -> false
+;;
+
+let isDigit = function
+  | '0' .. '9' -> true
+  | _ -> false
+
+let isAlpha = function
+  | 'a' .. 'z' | 'A' .. 'Z' -> true
+  | _ -> false
+
+(* Writing these out, instead of calling [is_alpha] and [is_digit], reduces
+   runtime by approx. 30% *)
+let isAlphanum = function
```

Comment:
Haskell has `isAlphaNum`, but alphanumeric is really a single word. C# has `isLetterOrDigit` (and `isLetter` instead of `isAlpha`). I think I prefer the C# naming.